### PR TITLE
Add New Session Management & Login/Logout

### PR DESCRIPTION
**Pull request by skyeslattery**
## Overview
Switched from user session system with session tokens system to device tokens. Allows for users to use favoriting system without being logged in. 

## Changes Made
Changed user_session app to device_token app. Changed login/logout routes accordingly.

## Test Coverage
Tested locally, needs to be tested with frontend to ensure correct behavior.

### CODE REVIEW

**Review by tjvignos — APPROVED** [review]
lgtm

File: src/user/views.py [hunk 1]
```diff
@@ -64,79 +63,134 @@ def remove_favorite_item(self, request, pk=None):
 
     @action(detail=False, methods=["post"], url_path="login")
     def login(self, request):
+        device_token = request.data.get("device_token")
```
Comment (Aayush-Agnihotri) [inline src/user/views.py:66]: device_token is snakecase but id_token_str is camelcase - might be weird for frontend to remember

**Review by Aayush-Agnihotri — APPROVED** [review]
lgtm

File: src/user/views.py (hunk 1 above)
Comment (skyeslattery) [reply src/user/views.py:66]: oh yeah i forgot to push to fix that originally

**Pull request merged**